src/runtime/Types/ManagedType.cs

```diff
@@ -83,8 +83,8 @@ internal static unsafe void DecrefTypeAndFree(StolenReference ob)
 
             var freePtr = Util.ReadIntPtr(type, TypeOffset.tp_free);
             Debug.Assert(freePtr != IntPtr.Zero);
-            var free = (delegate* unmanaged[Cdecl]<StolenReference, void>)freePtr;
-            free(ob);
+            var free = (delegate* unmanaged[Cdecl]<ref StolenReference, void>)freePtr;
+            free(ref ob);
```

Comment:
I think this is wrong. It does not say anywhere but I'd expect `tp_free` to take a reference to `PyObject` (e.g. `PyObject*`), not a reference to reference to it (e.g. not a `PyObject**`).

Reply:
Ah, that's probably one reason for the crashes :)